Enable UX to request fluxv2 available packages

absoludity opened the pull request:
### Description of the change

This will be the first in a list of PRs required to use the flux plugin from the UI (I'll add and link an issue to track them).

Running the Kubeapps UX with the fluxv2 plugin enabled results in:

![kubeapps-with-fluxv2-unimplemented](https://user-images.githubusercontent.com/497518/137844637-4da87477-809e-469b-a97c-4a5bc068fe5f.png)

This is because currently the plugin handles the case where the `request.Context.Cluster == ""` (ie. cluster isn't specified) only, in which case it correctly defaults to the cluster on which Kubeapps is installed. Since the UI will not know which requests plugins can handle (ie. which plugins support requests for different clusters etc.), it will always set the full context and rely on the plugin to respond with unimplemented or a valid response accordingly.

With this change, the response correctly comes back as empty rather than unimplemented (before I've added any flux `HelmRepositories`):

![kubeapps-with-fluxv2-empty](https://user-images.githubusercontent.com/497518/137845266-a36f524e-c108-4c5e-a568-ed14e1b1206c.png)

### Benefits

Don't see an error when browsing for available apps using the flux plugin.

### Possible drawbacks

Can't think of any.

### Applicable issues

<!-- Enter any applicable Issues here (You can reference an issue using #) -->
Ref: #3610 

### Additional information

<!-- If there's anything else that's important and relevant to your pull
request, mention that information here.-->

gfichtenholt reviewed (commented):
What about `newHelmActionConfigGetter()` in utils.go? Should that be also modified to be consistent?

absoludity commented:
> What about `newHelmActionConfigGetter()` in utils.go? Should that be also modified to be consistent?

Yes, and all the other operations will need to be updated. I've just written the tests and updated the one operation here because it enables me to continue focusing on the UX (assuming that's what you'd prefer), and have added an item to the related issue #3610  for doing the other operations. Let me know if you're able to do that item (no UX, plugin only) and we'll be able to get more UX work done.

gfichtenholt commented:
> > What about `newHelmActionConfigGetter()` in utils.go? Should that be also modified to be consistent?
> 
> Yes, and all the other operations will need to be updated. I've just written the tests and updated the one operation here because it enables me to continue focusing on the UX (assuming that's what you'd prefer), and have added an item to the related issue #3610 for doing the other operations. Let me know if you're able to do that item (no UX, plugin only) and we'll be able to get more UX work done.

ok, fine with me. Regarding other work. Generally, I am currently knee deep in #3496 (it is very hard and going slow), but if there a simple change I can make to unblock someone else, I am down

gfichtenholt reviewed (approved):
+1
